Wox.Core/Configuration/Portable.cs

```diff
@@ -0,0 +1,200 @@
+using Wox.Plugin.SharedCommands;
+
+namespace Wox.Core.Configuration
+{
+    public class Portable : IPortable
+    {
+        private UpdateManager portabilityUpdater;
+
+        public void DisablePortableMode()
+        {
+            portabilityUpdater = new UpdateManager(string.Empty, Constant.Wox, Constant.RootDirectory);
+
+            try
+            {
+                MoveUserDataFolder(DataLocation.PortableDataPath, DataLocation.RoamingDataPath);
+#if DEBUG
+                // Create shortcuts and uninstaller are not required in debug mode, 
+                // otherwise will repoint the path of the actual installed production version to the debug version
+#else
+                CreateShortcuts();
+                CreateUninstallerEntry();
+#endif
+                IndicateDeletion(DataLocation.PortableDataPath);
+
+                MessageBox.Show("Wox needs to restart to finish disabling portable mode, " +
+                    "after the restart your portable data profile will be deleted and roaming data profile kept");
+
+                portabilityUpdater.Dispose();
+
+                UpdateManager.RestartApp();
+            }
+            catch (Exception e)
+            {
+#if DEBUG
+                portabilityUpdater.Dispose();
+                throw;
+#else
+                portabilityUpdater.Dispose();
+                Log.Exception("Portable", "Error occured while disabling portable mode", e);
+                throw;
+#endif
+            }
+        }
+
+        public void EnablePortableMode()
+        {
+            portabilityUpdater = new UpdateManager(string.Empty, Constant.Wox, Constant.RootDirectory);
+
+            try
+            {
+                MoveUserDataFolder(DataLocation.RoamingDataPath, DataLocation.PortableDataPath);
+#if DEBUG
+                // Remove shortcuts and uninstaller are not required in debug mode, 
+                // otherwise will delete the actual installed production version
+#else
+                RemoveShortcuts();
+                RemoveUninstallerEntry();
+#endif
+                IndicateDeletion(DataLocation.RoamingDataPath);
+
+                MessageBox.Show("Wox needs to restart to finish enabling portable mode, " +
+                    "after the restart your roaming data profile will be deleted and portable data profile kept");
+
+                portabilityUpdater.Dispose();
+
+                UpdateManager.RestartApp();
+            }
+            catch (Exception e)
+            {
+#if DEBUG
+                portabilityUpdater.Dispose();
+                throw;
+#else
+                portabilityUpdater.Dispose();
+                Log.Exception("Portable", "Error occured while enabling portable mode", e);
+                throw;
+#endif
+            }
+        }
+
+        public void RemoveShortcuts()
+        {
+            var exeName = Constant.Wox + ".exe";
+            portabilityUpdater.RemoveShortcutsForExecutable(exeName, ShortcutLocation.StartMenu);
```

Comment:
since portability updater is created at specific methods currently this could be null (+ it is also disposed at these methods it gets an exception) so I think just moving this to private will solve issues

Reply:
Well I want the portability contract to include removing/creating shortcuts as a necessary part of making Wox portable. 

In the case that the method gets called on its on, what do you think a simple if statement to check null on the update manager and new up if doesn't exist and dispose when done, else run as per normal?

Reply:
Updated now, see [commit](https://github.com/jjw24/Wox/pull/143/commits/86903e5e63dc55d67c0e8a3c58aef846ee862a27)